src/main/java/hudson/remoting/Engine.java

```diff
@@ -350,6 +351,10 @@ public void setWebSocket(boolean webSocket) {
         this.webSocket = webSocket;
     }
 
+    public void setWebSocketHeaders(@Nonnull Map<String, String> webSocketHeaders) {
```

Comment:
Javadoc would be nice, because this endpoint might be used on both server and client sides by API users

Reply:
@oleg-nenashev added.